lib/headings.js

```diff
@@ -19,22 +19,27 @@ var headings = module.exports = function (md, options) {
 
   var savedTokenConstructor = false
   var slugger = new GithubSlugger()
+  var Token;
 
   md.core.ruler.push('githubHeadings', function (state) {
     // save the Token constructor because we'll be building a few instances at render
     // time; that's sort of outside the intended markdown-it parsing sequence, but
     // since we have tight control over what we're creating (a link), we're safe
     if (!savedTokenConstructor) {
       tokenUtil.set(state.Token)
+      Token = state.Token
       savedTokenConstructor = true
     }
   })
```

Comment:
It occurred to me while doing something else that we can eliminate `savedTokenConstructor` from this file and just do
```js
if (!Token) {
  Token = state.Token
  tokenUtil.set(Token)
}
```